markdown_xblock/html.py

```diff
@@ -2,53 +2,78 @@
 from xblock.fragment import Fragment
 from xblockutils.resources import ResourceLoader
 from xblockutils.studio_editable import StudioEditableXBlockMixin, loader
+from xblockutils.settings import XBlockWithSettingsMixin
 
-from .bleaching import SanitizedText
 from .utils import _
 
 log = logging.getLogger(__name__)  # pylint: disable=invalid-name
 xblock_loader = ResourceLoader(__name__)  # pylint: disable=invalid-name
 
-
-class HTML5XBlock(StudioEditableXBlockMixin, XBlock):
+SETTINGS_KEY = 'markdown'
+DEFAULT_EXTRAS = [
+    "code-friendly",
+    "fenced-code-blocks",
+    "footnotes",
+    "tables",
+    "use-file-vars"
+]
+DEFAULT_SETTINGS = {
+    "extras": DEFAULT_EXTRAS
+}
+
+
+def get_xblock_settings():
+    """Extract xblock settings."""
+    try:
+        xblock_settings = django_settings.XBLOCK_SETTINGS
+    except AttributeError:
+        settings = DEFAULT_SETTINGS
+    else:
+        settings = xblock_settings.get(
+            SETTINGS_KEY, DEFAULT_SETTINGS)
+    return settings
+
+
+@XBlock.wants('settings')
+class MarkdownXBlock(StudioEditableXBlockMixin, XBlockWithSettingsMixin, XBlock):
     """
-    This XBlock will provide an HTML WYSIWYG interface in Studio to be rendered in LMS.
+    This XBlock provides content editing in Markdown and displays it in HTML.
     """
 
     display_name = String(
         display_name=_('Display Name'),
         help=_('The display name for this component.'),
         scope=Scope.settings,
-        default=_('Text')
+        default=_('Markdown')
     )
-    data = String(help=_('Html contents to display for this module'), default=u'', scope=Scope.content)
-    allow_javascript = Boolean(
-        display_name=_('Allow JavaScript execution'),
-        help=_('Whether JavaScript should be allowed or not in this module'),
-        default=False,
+    data = String(
+        help=_('The Markdown content for this module'),
+        default=u'',
         scope=Scope.content
     )
-    editor = String(
-        help=_(
-            'Select Visual to enter content and have the editor automatically create the HTML. Select Raw to edit '
-            'HTML directly. If you change this setting, you must save the component and then re-open it for editing.'
-        ),
-        display_name=_('Editor'),
-        default='visual',
-        values=[
-            {'display_name': _('Visual'), 'value': 'visual'},
-            {'display_name': _('Raw'), 'value': 'raw'}
-        ],
-        scope=Scope.settings
+    allow_inline_html = Boolean(
+        display_name=_('Allow inline HTML (Admin only)'),
+        help=_('Whether inline HTML should be allowed or not in this module'),
+        default=False,
+        scope=Scope.content
     )
-    editable_fields = ('display_name', 'editor', 'allow_javascript')
+    editor = 'markdown'
+
+    @property
+    def editable_fields(self):
+        """The property for editable fields."""
+        if self.runtime._user.is_staff:
```

Comment:
I'm not sure I like this bit. This appears to allow a `staff` user to decide whether inline HTML is allowed or not. Staff users and course authors are often identical groups. That means that I as a site operator can't decide whether I trust course authors to not inject malicious or broken code; it's course authors that decide for themselves. Which makes the toggle kind of useless.

So I'd rather see this set via `settings` (which the site operator, not a course author, controls). Or have a toggle in `settings` which defines whether staff users can set this.

Reply:
Ok, that makes sense, I'll do that.